### PR TITLE
Add project function to camera and fix cocos ui bug

### PR DESCRIPTION
**Pull request by super626**
1. add project function to camera
2. Fix bug that camera mask does not work for cocos ui

### CODE REVIEW

**Comment by minggo** [comment]
Should modify corresponding lua test case.

File: cocos/2d/CCCamera.h [hunk 1]
```diff
@@ -130,11 +130,30 @@ class CC_DLL Camera :public Node
...
-    * Convert the specified point of viewport from screenspace coordinate into the worldspace coordinate.
-    */
-    void unproject(const Size& viewport, Vec3* src, Vec3* dst) const;
+     * Convert the specified point of viewport from screen-space coordinate into the world-space coordinate.
+     *
+     * @param viewport The viewport size to use.
+     * @param src The screen-space position.
+     * @param dst The world-space position.
+     */
+    CC_DEPRECATED_ATTRIBUTE void unproject(const Size& viewport, const Vec3* src, Vec3* dst) const;
```
Comment (minggo) [inline cocos/2d/CCCamera.h]: Should use `CC_DEPRECATED(v3.5)`

**Comment by minggo** [comment]
Why we modify the API?
As the [document](https://github.com/cocos2d/cocos2d-x/blob/v3/docs/RELEASE_ENGINEERING.md#deprecated-apis) said, we can not deprecated API in minor versions.

**Comment by darkdukey** [comment]
Sorry, Tony modified the API based on my comments. But I feel the new API is more user friendly
before

``` cpp
Vec3 screenPosition(0,0,0)
Vec3 worldPosition;
camera->unproject(Director::getInstance()->getWinSize(), &screenPosition, &worldPosition)
```

after

``` cpp
Vec3 screenPosition(0,0,0)
Vec3 worldPosition = camera->unprotect(&screenPosition)
```

Maybe we should use reference
so we can do 

``` cpp
Vec3 screenPosition(0,0,0)
Vec3 worldPosition = camera->unprotect(screenPosition)
```

File: cocos/2d/CCCamera.h [hunk 2]
```diff
@@ -126,11 +126,30 @@ class CC_DLL Camera :public Node
 
     /**get view projection matrix*/
     const Mat4& getViewProjectionMatrix() const;
+    
+    /* convert the specified point of viewport from world-space coordinates into the screen-space coordinates.
+     *
+     * @param src The world-space position.
+     * @return The screen-space position.
+     */
+    Vec2 project(const Vec3* src) const;
```
Comment (darkdukey) [inline cocos/2d/CCCamera.h]: How about `Vec2 project(const Vec3& src) const;`

File: cocos/2d/CCCamera.h [hunk 3]
```diff
@@ -126,11 +126,30 @@ class CC_DLL Camera :public Node
...
+    
+    /**
+     * Convert the specified point of viewport from screen-space coordinate into the world-space coordinate.
+     *
+     * @param src The screen-space position.
+     * @return The world-space position.
+     */
+    Vec3 unproject(const Vec3* src) const;
```
Comment (darkdukey) [inline cocos/2d/CCCamera.h]: How about 

``` cpp
Vec3 unproject(const Vec2& src)
Vec3 unproject(const Vec3& src)
```

**Comment by minggo** [comment]
Yep, i think using reference is better is possible.

**Comment by super626** [comment]
@darkdukey as minggo said,
According to RELEASE_ENGINEERING.md, we can not deprecated api in this version.

**Comment by minggo** [comment]
Yep.
We can do it in v4-develop but not v3.5.

**Comment by super626** [comment]
Had removed deprecated

**Comment by minggo** [comment]
Please modify changelog, and send a pull request to v4-develop too.
Thanks.

**Comment by super626** [comment]
fix #10449

**Pull request merged**